esp-hal/src/timer/timg.rs

```diff
@@ -532,6 +531,27 @@ impl Timer {
             .t(self.timer)
             .bit_is_set()
     }
+
+    fn set_interrupt_enabled(&self, state: bool) {
+        cfg_if::cfg_if! {
```

Comment:
maybe we want to keep and extend the original comment (`// always use level interrupt`) - i.e. why we need level interrupt on ESP32/S2?